app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHours.kt

```diff
@@ -22,7 +22,7 @@ class AddOpeningHours(
 
     /* See also AddWheelchairAccessBusiness and AddPlaceName, which has a similar list and is/should
        be ordered in the same way for better overview */
-    private val filter by lazy { ("""
```

Comment:
dropping private here seems smelly to me, but was needed for tests

Reply:
Yes, please don't.

Why don't you use `isApplicableTo`?

Reply:
1)

I wanted to test only filter here (as it is part that I modified). But likely testing entire actually run function makes more sense

2)

`isApplicableTo` is also checking `hasName` which calls `hasFeatureName`

`hasFeatureName` relies on `featureDictionaryFuture` which is just `mock()` - and as result tests were crashing with some esoteric stacktrace . Though there are some exsisting similar tests that actually work, I will look into it (if you are not fundamentally opposed to basic idea of this quest)

Reply:
No I am not opposed. You can configure a mock to return what you want. Anyway, I think `hasFeatureName` is only called if the feature itself has no name, so you can just work around it by adding `name` as a tag.

Reply:
Well, I wanted to test that it will work also for some specific unnamed features.

I will configure mock then instead of visibility hacks.

Reply:
Though looking at

```
    private fun hasFeatureName(tags: Map<String, String>): Boolean =
        featureDictionaryFuture.get().byTags(tags).isSuggestion(false).find().isNotEmpty()
```

it seems that I would need mock returning in `get()` a mock that would be returning another mock in `isSuggestion` and so on.

Building own `FeatureDictionary` for this single test also seems supremely annoying.

I think I would rather drop that test.